spec/granite/migrator/migrator_spec.cr

```diff
@@ -0,0 +1,135 @@
+require "../../spec_helper"
+
+describe Granite::Migrator::Base do
+  describe "#drop_sql" do
+    it "generates correct SQL with #{{{ env("CURRENT_ADAPTER") }}} adapter" do
+      {% if env("CURRENT_ADAPTER") == "mysql" %}
+        Review.migrator.drop_sql.should eq "DROP TABLE IF EXISTS `reviews`;"
+      {% else %}
+        Review.migrator.drop_sql.should eq "DROP TABLE IF EXISTS \"reviews\";"
+      {% end %}
+    end
+  end
+
+  describe "#create_sql" do
+    it "generates correct SQL with #{{{ env("CURRENT_ADAPTER") }}} adapter" do
+      {% if env("CURRENT_ADAPTER") == "pg" %}
+        Review.migrator.create_sql.should eq <<-SQL
+          CREATE TABLE "reviews"(
+          "id" BIGSERIAL PRIMARY KEY,
+          "name" VARCHAR(255)
+          ,
+          "downvotes" INT
+          ,
+          "upvotes" BIGINT
+          ,
+          "sentiment" FLOAT
+          ,
+          "interest" REAL
+          ,
+          "published" BOOL
+          ,
+          "created_at" TIMESTAMP
+          ) ;\n
+          SQL
+
+        Kvs.migrator.create_sql.should eq <<-SQL
+          CREATE TABLE "kvs"(
+          "k" VARCHAR(255) PRIMARY KEY,
+          "v" VARCHAR(255)
+          ) ;\n
+          SQL
+
+        UUIDModel.migrator.create_sql.should eq <<-SQL
+          CREATE TABLE "uuids"(
+          "uuid" UUID PRIMARY KEY) ;\n
+          SQL
+
+        # Also check Array types for pg
+        ArrayModel.migrator.create_sql.should eq <<-SQL
+          CREATE TABLE "array_model"(
+          "id" SERIAL PRIMARY KEY,
+          "str_array" TEXT[]
+          ,
+          "i16_array" SMALLINT[]
+          ,
+          "i32_array" INT[]
+          ,
+          "i64_array" BIGINT[]
+          ,
+          "f32_array" REAL[]
+          ,
+          "f64_array" DOUBLE PRECISION[]
+          ,
+          "bool_array" BOOLEAN[]
+          ) ;\n
+          SQL
+
+      {% elsif env("CURRENT_ADAPTER") == "mysql" %}
+        Review.migrator.create_sql.should eq <<-SQL
+          CREATE TABLE `reviews`(
+          `id` BIGINT NOT NULL AUTO_INCREMENT PRIMARY KEY,
+          `name` VARCHAR(255)
+          ,
+          `downvotes` INT
+          ,
+          `upvotes` BIGINT
+          ,
+          `sentiment` FLOAT
+          ,
+          `interest` REAL
```

Comment:
Can make this `DOUBLE` based on [this chart](http://www.peachpit.com/articles/article.aspx?p=1752305&seqNum=2)